src/main/java/core/basesyntax/Rectangle.java

```diff
@@ -27,4 +27,26 @@ public Integer getLength() {
     public String getColor() {
         return color;
     }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) {
+            return true;
+        }
+        if (o == null || getClass() != o.getClass()) {
+            return false;
+        }
+        Rectangle rectangle = (Rectangle) o;
+        return (width != null ? !width.equals(rectangle.width) : rectangle.width != null)
+                || (length != null ? !length.equals(rectangle.length) : rectangle.length != null)
+                || (color != null ? color.equals(rectangle.color) : rectangle.color == null);
```

Comment:
let's refactor this
for each field, we should return `true` 
if `this.field` and `rectangle.field` are both null 
OR if `this.field` is not null AND is equal to `rectangle.field`